src/libserver/milter.c

```diff
@@ -1671,11 +1675,14 @@ rspamd_milter_remove_header_safe (struct rspamd_milter_session *session,
 				rspamd_milter_send_action (session,
 						RSPAMD_MILTER_CHGHEADER,
 						ar->len + nhdr + 1, hname, hvalue);
+				priv->cur_hdr --;
 			}
 		}
 
 		g_string_free (hname, TRUE);
 		g_string_free (hvalue, TRUE);
+
+		g_assert (priv->cur_hdr >= 0);
```

Comment:
While it seems correct, I would suggest to replace it with an `msg_err`. The reason why is that we can get any potentially malformed milter block from the Lua code, but that should not trigger assertions!

Reply:
OK, I'll re-push the change.